semgrep/semgrep/core_runner.py

```diff
@@ -26,10 +29,55 @@
 from semgrep.semgrep_types import OPERATORS
 from semgrep.util import debug_print
 from semgrep.util import INVALID_PATTERN_EXIT_CODE
+from semgrep.util import partition
 from semgrep.util import print_error
 from semgrep.util import print_error_exit
 
 
+def get_re_matches(patterns: Dict, path: Path) -> List[PatternMatch]:
+    contents = path.read_text()
+
+    return [
+        PatternMatch(
+            {
+                "check_id": pattern["id"],
+                "path": str(path),
+                "start": {"offset": match.start()},
+                "end": {"offset": match.end()},
+                "extra": {"lines": [contents[match.start() : match.end()]]},
+            }
+        )
+        for pattern in patterns
+        for match in re.finditer(pattern["pattern"], contents)
+    ]
+
+
+def get_target_files(
+    targets: List[Path], exclude: List[str], include: List[str]
+) -> List[Path]:
+    if not include:
+        # Default to all files
+        include = ["*"]
+
+    filepaths = [
+        target
+        for target in targets
+        if target.is_file()
+        and any(target.match(i) for i in include)
```

Comment:
Just FYI re. fnmatch if you haven't seen it before (not sure if it uses the same syntax as our include / exclude).

Reply:
Looks like they are similar, but not the same:

- `semgrep-core` include/exclude: https://dune.readthedocs.io/en/stable/concepts.html#glob
- `semgrep` include/exclude: https://docs.python.org/3/library/fnmatch.html

From what I can tell, `semgrep-core` globbing looks like a superset of `fnmatch`'s functionality. The Python [`glob`](https://docs.python.org/3/library/glob.html) module seems closer to the OCaml version, although still not exactly the same. The `rglob` call uses `glob` under the hood, so our targets are generated similarly to the OCaml version, but include/exclude currently operates via `fnmatch` as you mentioned, so there's some divergence from the OCaml version.

Probably not worth our time to completely unify the behaviors because 1.) I imagine 95% of include/exclude will be something like `*.py` or `*.js`, which OCaml glob handles the same as `fnmatch`, and 2.) we'd likely have to re-implement the OCaml glob code to completely replicate it's behavior - I can't find any standard lib modules that do so.

Thoughts?

Reply:
> Probably not worth our time to completely unify the behaviors because 1.) I imagine 95% of include/exclude will be something like *.py or *.js

Yeah we can fix once we have some idea of what people's use cases are and adjust from there